packages/message/src/lib/__tests__/message-service.spec.ts

```diff
@@ -0,0 +1,305 @@
+import { eventBus } from '@adorsys-gis/event-bus';
+import {
+  ServiceResponse,
+  ServiceResponseStatus,
+} from '@adorsys-gis/status-service';
+import { v4 as uuidv4 } from 'uuid';
+import { Message } from '../../model/Message';
+import { MessageEventChannel } from '../../model/MessageEventChannel';
+import { MessageService } from '../MessageService';
+
+describe('MessageService', () => {
+  let messageService: MessageService;
+
+  beforeEach(() => {
+    // Initialize the MessageService before each test
+    messageService = new MessageService(eventBus);
+  });
+
+  afterEach(async () => {
+    // Clear all messages after each test
+    const deleteAllMessagesEvent = new Promise<void>((resolve) => {
+      eventBus.once(
+        MessageEventChannel.GetAllByContactId,
+        async (response: ServiceResponse<Message[]>) => {
+          if (response.status === ServiceResponseStatus.Success) {
+            const messages = response.payload;
+            if (Array.isArray(messages)) {
+              for (const message of messages) {
+                messageService.deleteMessage(message.id!);
+              }
+            }
+          }
+          resolve();
+        },
+      );
+
+      messageService.getAllMessagesByContact('did:key:zKAJCbjwdhuhJBJWHBDSIs');
+    });
+
+    await deleteAllMessagesEvent;
+  });
+
+  // Helper function to wait for an event
+  const waitForEvent = (channel: MessageEventChannel) => {
+    return new Promise<ServiceResponse<Message | Message[]>>((resolve) => {
+      eventBus.once(channel, (data) => resolve(data));
+    });
+  };
+
+  it('should create a new message and emit the event', async () => {
+    const newMessage: Message = {
+      id: uuidv4(),
+      text: 'Hello, how are you',
+      sender: 'did:key:z92389jqjdNJAWOJNSWDDjies',
+      contactId: 'did:key:zKAJCbjwdhuhJBJWHBDSIs',
+      timestamp: new Date(),
+    };
+
+    const createEvent = waitForEvent(MessageEventChannel.CreateMessage);
+    messageService.createMessage(newMessage);
+
+    const createdMessage = await createEvent;
+
+    expect(createdMessage).toEqual(
+      expect.objectContaining({
+        status: ServiceResponseStatus.Success,
+        payload: expect.objectContaining({
+          id: newMessage.id,
+          text: 'Hello, how are you',
+          sender: 'did:key:z92389jqjdNJAWOJNSWDDjies',
+          contactId: newMessage.contactId,
+          timestamp: newMessage.timestamp,
+        }),
```

Comment:
Any benefit over this?

```suggestion
        payload: newMessage,
```